src/Game.js

```diff
@@ -0,0 +1,101 @@
+const Frame = require("./frame");
+
+module.exports = class Game {
+  constructor(frameClass = Frame) {
+    this.frames = [];
+    this.frameClass = frameClass;
+    this.totalScore = 0;
+    this.addNewFrame();
+    this.MAX_FRAMES = 10;
+  }
+
+  numOfFrames() {
+    return this.frames.length;
+  }
+
+  addNewFrame() {
+    this.frames.push(new this.frameClass());
+  }
+
+  lastFrame() {
+    return this.frames[this.frames.length - 1];
+  }
+
+  previous_frame() {
+    if (!this.frames.length < 2) {
+      return this.frames[this.frames.length - 2];
+    }
+  }
+
+  next_frame(index) {
+    return this.frames[index + 1];
+  }
+
+  next_to_next_frame(index) {
+    return this.frames[index + 2];
+  }
+
+  isNinthFrame(index) {
+    return index === this.MAX_FRAMES - 2;
+  }
+
+  strikeBonus(index) {
+    if (this.isFinalFrame(index)) {
+      return 0;
+    } else if (this.isNinthFrame(index)) {
+      return this.next_frame(index).firstTwoRolls();
+    } else {
+      if (this.next_frame(index).isStrike()) {
+        return (
+          this.next_frame(index).frameScore() +
+          this.next_to_next_frame(index).firstRoll()
+        );
+      } else {
+        return this.next_frame(index).frameScore();
+      }
+    }
+  }
```

Comment:
there are some long functions which are difficult to follow

Reply:
also I think it is easier and more object-oriented to have the frame decide if it deserves a bonus, but that's up for debate